src/Migration/Reader/Map.php

```diff
@@ -211,16 +210,15 @@ public function isFieldMapped($document, $field, $type)
      */
     public function getDocumentMap($document, $type)
     {
-        $this->validateType($type);
+        $key = $document . '-' . $type;
+        if (isset($this->documentsMap[$key])) {
+            return $this->documentsMap[$key];
+        }
```

Comment:
If the result is already present in the `$this-> documentsMap`, then calculation can be omitted